internal/statusupdater/statusupdater.go

```diff
@@ -1,24 +1,29 @@
-package module
+package statusupdater
 
 import (
 	"context"
+	"time"
 
 	ootov1alpha1 "github.com/qbarrand/oot-operator/api/v1alpha1"
 	"github.com/qbarrand/oot-operator/internal/daemonset"
 	"github.com/qbarrand/oot-operator/internal/metrics"
 	appsv1 "k8s.io/api/apps/v1"
 	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-//go:generate mockgen -source=statusupdater.go -package=module -destination=mock_statusupdater.go
+//go:generate mockgen -source=statusupdater.go -package=statusupdater -destination=mock_statusupdater.go
 
 type StatusUpdater interface {
-	UpdateModuleStatus(ctx context.Context,
-		mod *ootov1alpha1.Module,
-		kernelMappingNodes []v1.Node,
-		targetedNodes []v1.Node,
-		dsByKernelVersion map[string]*appsv1.DaemonSet) error
+	UpdateModuleStatus(ctx context.Context, mod *ootov1alpha1.Module, kernelMappingNodes []v1.Node,
+		targetedNodes []v1.Node, dsByKernelVersion map[string]*appsv1.DaemonSet) error
+	PresetPreflightVerificationStatus(ctx context.Context, pv *ootov1alpha1.PreflightValidation,
+		status *ootov1alpha1.CRStatus) error
+	SetPreflightVerificationStatus(ctx context.Context, preflight *ootov1alpha1.PreflightValidation, status *ootov1alpha1.CRStatus,
+		verificationStatus string, message string) error
+	SetPreflightVerificationStage(ctx context.Context, preflight *ootov1alpha1.PreflightValidation,
+		status *ootov1alpha1.CRStatus, stage string) error
```

Comment:
How about having the `StatusUpdater` interface to consist of 2 (3 in the future) private embedded interfaces for clear separation of concern?
Something like:
```
type StatusUpdated interface {
    moduleStatusUpdater
    preflightStatusUpdater
}
```

If you don't like it, how about at least have a better prefix naming?
```
ModuleUpdateStatus(...) error
PreflightGetVerificationStatus(...) error
PreflightSetVerificationStatus(...) error
PreflightSetVerificationStage(...) 
```
?

Reply:
I don't see a reason to define additional interface that we are not going to use anyway. I will change the prefixes as suggested